cmake/OmimHelpers.cmake

```diff
@@ -2,8 +2,9 @@
 set(OMIM_WARNING_FLAGS
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wall>
   $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wextra>
-  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wno-unused-parameter>
-  $<$<CXX_COMPILER_ID:AppleClang>:-Wno-deprecated-declarations>  # boost warnings
+  $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:-Wpedantic>
```

Comment:
Does it work on all our checks well? Is the output clean for iOS, Android, Mac, Linux GCC and Linux Clang?

Reply:
Yes, all outputs are clean except for Linux GCC. It was messed up before and we should clean it first to enable `-Werror` for it